x-pack/plugins/fleet/server/services/output.ts

```diff
@@ -952,6 +953,31 @@ class OutputService {
       }
     }
   }
+
+  async getLatestOutputHealth(esClient: ElasticsearchClient, id: string): Promise<OutputHealth> {
+    const response = await esClient.search({
+      index: OUTPUT_HEALTH_DATA_STREAM,
+      query: { bool: { filter: { term: { output: id } } } },
+      sort: { '@timestamp': 'desc' },
+      size: 1,
+    });
+    if (response.hits.hits.length === 0) {
+      return {
+        state: 'UNKOWN',
+        message: '',
+      };
+    }
+    const latestHit = response.hits.hits[0]._source as any;
+    return {
+      state: latestHit.state,
+      message: latestHit.message ?? '',
```

Comment:
we might want to add the timestamp, to report the last time on the UI, in case the health reporting stopped and the state might be stale

Reply:
Added a tooltip with the last reported time:
<img width="1103" alt="image" src="https://github.com/elastic/kibana/assets/90178898/d01f5990-c5e2-4fca-92c7-343d72951a51">